src/components/graph/ExerciseTreeGraph.tsx

```diff
@@ -0,0 +1,496 @@
+import React from 'react'
+import * as d3 from 'd3'
+
+import { ExerciseGraph } from './ExerciseGraph'
+import { ExerciseGraphLayout } from './ExerciseGraphLayout'
+
+import { CheckOrCross } from '../CheckOrCross'
+
+import { slug, position } from './graph-types'
+
+type ExerciseTreeGraphProps = {
+  config?: TrackConfiguration
+}
+
+export class ExerciseTreeGraph extends React.Component<ExerciseTreeGraphProps> {
+  private graphRef = React.createRef<HTMLDivElement>()
```

Comment:
```suggestion
  private graphRef = React.createRef<HTMLDivElement | null>(null)
```

My guess is that this is the correct type for this, because the ref is `null` before it's set. (It's actually undefined, but whatever).

Reply:
:shrug: I just followed this: https://reactjs.org/docs/refs-and-the-dom.html and there is no null arguement there.  I can add it

Reply:
Your suggestion errors
```
/home/tim/Projects/neenjaw-exercism/v3-dashboard/src/components/graph/ExerciseTreeGraph.tsx
TypeScript error in /home/tim/Projects/neenjaw-exercism/v3-dashboard/src/components/graph/ExerciseTreeGraph.tsx(17,62):
Expected 0 arguments, but got 1.  TS2554

    15 | 
    16 | export class ExerciseTreeGraph extends React.Component<ExerciseTreeGraphProps> {
  > 17 |   private graphRef = React.createRef<HTMLDivElement | null >(null)
       |                                                              ^
    18 | 
    19 |   public shouldComponentUpdate(): boolean {
    20 |     return false
```